amads/melody/contour/interpolation_contour.py

```diff
@@ -0,0 +1,272 @@
+"""
+Calculates the Interpolation Contour of a melody, along with related features, as
+implemented in the FANTASTIC toolbox of Müllensiefen (2009).
+"""
+
+__author__ = "David Whyatt"
+
+import numpy as np
+
+
+class InterpolationContour:
+    """Class for calculating and analyzing interpolation contours of melodies."""
+
+    def __init__(self, times: list[float], pitches: list[int]):
+        """Initialize with time and pitch values.
+
```

Comment:
Any particular reason why InterpolationContour takes onset times but StepContour takes durations?

Reply:
As above, maybe we can improve on this implementation a bit